# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Eman123-123/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why
Lane: Content Refresh Prioritization
I want to study which existing pages may be useful candidates for content refresh. The decision is whether a page should be prioritized for review and possible refresh. This lane is useful because the data contains search visibility, traffic, engagement, page age, and freshness signals. I will use these observed signals to support prioritization rather than claim that a refresh will definitely improve performance.
*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [16]:
!git clone https://github.com/Eman123-123/flyrank-ml-internship.git /content/flyrank-ml-internship

fatal: destination path '/content/flyrank-ml-internship' already exists and is not an empty directory.


In [17]:
import os

os.chdir("/content/flyrank-ml-internship")

print("Current folder:", os.getcwd())
print("\nFiles:")
print(os.listdir())

Current folder: /content/flyrank-ml-internship

Files:
['data', 'GUIDE.md', 'notebooks', 'submission', 'skills', 'DATA_USE.md', '.github', '.git', 'scripts', 'SETUP.md', 'requirements.txt', 'CLAUDE.md', 'outputs', 'LICENSE', 'README.md', 'docs', 'AGENTS.md', '.gitignore', 'work']


In [18]:
import pandas as pd

path = "data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(path)

print("Data shape:", df.shape)
print("Average impressions:", round(df["impressions_90d"].mean(), 2))
print("Average CTR:", round(df["ctr"].mean(), 4))
print("Average content age (days):", round(df["content_age_days"].mean(), 2))

Data shape: (30000, 44)
Average impressions: 5200.37
Average CTR: 0.5107
Average content age (days): 256.17


In [19]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call
The decision is which existing pages should be prioritized for review and possible content refresh.

A content or SEO team could use the prioritization to decide which pages to review first and where to spend limited refresh resources.

A false positive could waste time and effort refreshing a page that does not need attention. A false negative could cause a potentially declining page to be missed, which may mean losing an opportunity to improve or protect its search visibility.

The goal is therefore decision-support: use observed page-level signals to help prioritize review, rather than automatically deciding that a page must be refreshed.
*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [20]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)
The starter dataset contains 30,000 content records and 44 columns.

Three observed numbers support the content-refresh prioritization lane:

- Average impressions over 90 days: 5,200.37
- Average CTR: 0.5107
- Average content age: 256.17 days

These numbers show that the dataset contains pages with measurable search visibility, click-through behavior, and substantial content age. This makes it reasonable to investigate whether existing pages can be prioritized for review using observed performance and freshness signals.
*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [21]:
import os

print("Current folder:", os.getcwd())

path = "data/raw/content_refresh_anonymized.csv"

print("Dataset exists:", os.path.exists(path))

Current folder: /content/flyrank-ml-internship
Dataset exists: True


## 4. Careful words: what I can and can't claim
### What I can claim

I can measure observed differences in search visibility, CTR, engagement, content age, and freshness across pages. I can use these signals to support a prioritization decision about which pages may deserve review.

### What I cannot claim

I cannot claim that a page will definitely improve after a content refresh. I also cannot claim that the observed relationships are causal. Any model developed later should be treated as decision-support and evaluated on held-out data before being used for prioritization.
*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [22]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.